### PR TITLE
bitnami/kafka Fix pem auth with custom encrypted private key

### PR DESCRIPTION
**Pull request by Voldemat**
### Description of the change

Also fixed and error with openssl when using pem auth with custom key. Just provided necessary password from env

### Benefits

Allows to use pem authentication with custom encrypted private key

### Possible drawbacks

None to my knowledge

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/kafka/templates/scripts-configmap.yaml [hunk 1]
```diff
@@ -192,15 +192,15 @@ data:
         if [[ -f "/mounted-certs/kafka-${POD_ROLE}-${POD_ID}.crt" && "/mounted-certs/kafka-${POD_ROLE}-${POD_ID}.key" ]]; then
           cp "/mounted-certs/kafka-${POD_ROLE}-${POD_ID}.crt" /certs/tls.crt
           # Copy the PEM key ensuring the key used PEM format with PKCS#8
-          openssl pkcs8 -topk8 -nocrypt -in "/mounted-certs/kafka-${POD_ROLE}-${POD_ID}.key" > /certs/tls.key
+          openssl pkcs8 -topk8 -nocrypt -passin pass:"${KAFKA_TLS_PEM_KEY_PASSWORD}" -in "/mounted-certs/kafka-${POD_ROLE}-${POD_ID}.key" > /certs/tls.key
```
Comment (migruiz4) [inline bitnami/kafka/templates/scripts-configmap.yaml:195]:
```suggestion
          openssl pkcs8 -topk8 -nocrypt -passin pass:"${KAFKA_TLS_PEM_KEY_PASSWORD:-}" -in "/mounted-certs/kafka-${POD_ROLE}-${POD_ID}.key" > /certs/tls.key
```

File: bitnami/kafka/templates/scripts-configmap.yaml [hunk 2]
```diff
@@ -192,15 +192,15 @@ data:
...
         elif [[ -f /mounted-certs/kafka.crt && -f /mounted-certs/kafka.key ]]; then
           cp "/mounted-certs/kafka.crt" /certs/tls.crt
           # Copy the PEM key ensuring the key used PEM format with PKCS#8
-          openssl pkcs8 -topk8 -nocrypt -in "/mounted-certs/kafka.key" > /certs/tls.key
+          openssl pkcs8 -topk8 -passin pass:"${KAFKA_TLS_PEM_KEY_PASSWORD}" -nocrypt -in "/mounted-certs/kafka.key" > /certs/tls.key
```
Comment (migruiz4) [inline bitnami/kafka/templates/scripts-configmap.yaml:199]:
```suggestion
          openssl pkcs8 -topk8 -passin pass:"${KAFKA_TLS_PEM_KEY_PASSWORD:-}" -nocrypt -in "/mounted-certs/kafka.key" > /certs/tls.key
```

File: bitnami/kafka/templates/scripts-configmap.yaml [hunk 3]
```diff
@@ -192,15 +192,15 @@ data:
...
         elif [[ -f /mounted-certs/tls.crt && -f /mounted-certs/tls.key ]]; then
           cp "/mounted-certs/tls.crt" /certs/tls.crt
           # Copy the PEM key ensuring the key used PEM format with PKCS#8
-          openssl pkcs8 -topk8 -nocrypt -in "/mounted-certs/tls.key" > /certs/tls.key
+          openssl pkcs8 -topk8 -passin pass:"${KAFKA_TLS_PEM_KEY_PASSWORD}" -nocrypt -in "/mounted-certs/tls.key" > /certs/tls.key
```
Comment (migruiz4) [inline bitnami/kafka/templates/scripts-configmap.yaml:203]:
```suggestion
          openssl pkcs8 -topk8 -passin pass:"${KAFKA_TLS_PEM_KEY_PASSWORD:-}" -nocrypt -in "/mounted-certs/tls.key" > /certs/tls.key
```

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Thank you for your contribution @Voldemat, but I'm afraid I cannot accept this contribution as it is.

While this may fix an issue when using password-protected PEM keys, it will then cause failures when non password-protected PEM keys are used.

I have suggested a small change that would fix this issue while preserving the both use cases.

**Comment by Voldemat** [comment]
@migruiz4 Added request changes and merged upstream.

**Review by migruiz4 — APPROVED** [review]
Thank you for applying the suggestions @Voldemat, looks good to me!

**Pull request merged**